src/ObsidianZoomPlugin.ts

```diff
@@ -140,12 +140,8 @@ export default class ObsidianZoomPlugin extends Plugin {
   }
 
   private isLegacyEditorEnabled() {
-    const config: { legacyEditor: boolean } = {
-      legacyEditor: true,
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      ...(this.app.vault as any).config,
-    };
-
-    return config.legacyEditor;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const config = (this.app.vault as any).config;
+    return (!("legacyEditor" in config) && true) || config.legacyEditor;
```

Comment:
It looks like [your version should work like the original](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABDAzgGQKYHMCGECeAogCYxRwBOhYOARgDYbEDCIFFGYUAFAG44h6UZgmAwsALkQ4w+AJSIA3gChEaxBAQooG0eKmLEjXARJlKU2nDiMZiAL6IAvEtXr3xvEVLkKUqBQgGAA0bu5qAHRR-ILCelih7vYA3MphiBxQbEiaYGJYEZ6mPpSp9mmgkLAIyOjYXma+1HSMxAByGADufAJCInn60rIKKu652roDWM6IMX3xqe6Z2YjcAITcAERF3uYUm8g58QoAZCeIAUEKAD7Xk-mF9cV7ZWnKjDoQwNMuiinK4xsGEKcCw3C+CVqmBMuyaNAYTFY7E4PAhcmCUKesMozQR7S64O+cjkaQhM0UO0aFkQwBw9BQGH+gMYILBEIxqGhDRKVHhrSRHC4hKw6MxMKpvJaTA63TRJIB33JlJ5-kCjNSzOB9FBwo5dXFPNx-LYgtRRL1XOecKl+NlRJJQA). What were you trying to change?

Reply:
Yeah this doesn't change anything. The issue is that the new version of obsidian does not include the setting at all unless you use it. Meaning it will be undefined and still the new editor is used. Took me quite some time figuring out why the extension wasn't working. So just change linwe 144 to`legacyEditor: false,`.

Reply:
Btw, toggling the setting once in the menu will acually create the setting. So this is how i fixed my installation. But everyone new will still have this problem.

Reply:
> Yeah this doesn't change anything. The issue is that the new version of obsidian does not include the setting at all unless you use it. Meaning it will be undefined and still the new editor is used. Took me quite some time figuring out why the extension wasn't working. So just change line 144 to`legacyEditor: false,`.

The problem is, if you change line 144, aren't you saying that Vaults that don't have that setting are set to false? But aren't Vaults without that setting from Obsidian versions when the legacy editor was the only one available?

Reply:
I think we should change line 144 to `legacyEditor: false` as @lukemt suggested. That should be enough to fix the issue.

Reply:
> The problem is, if you change line 144, aren't you saying that Vaults that don't have that setting are set to false? But aren't Vaults without that setting from Obsidian versions when the legacy editor was the only one available?

I know, the problem is, it means either one of two things:
- It is a very old version
- It is the new version
I beleave supporting the new version would be the right thing. Also keep in mind that the extension will be deactivated when the warning shows up.